src/payment/strategies/stripev3/stripev3-payment-strategy.ts

```diff
@@ -87,6 +87,10 @@ export default class StripeV3PaymentStrategy implements PaymentStrategy {
 
                 paymentIntent = await this._confirmVaultedPayment(clientSecret);
 
+                if (!paymentIntent.id) {
+                    throw new NotInitializedError(NotInitializedErrorType.PaymentNotInitialized);
```

Comment:
Usually,  `PaymentNotInitialized` is used when something goes wrong with the initialize step of the strategy.
A `MissingDataError` might be more accurate for this situation 🍹

Reply:
Thanks, which one do you think it suits bests?
**MissingDataErrorType.MissingPayment** or **MissingDataErrorType.MissingPaymentToken**